src/Console/Helper/TaskRunnerHelper.php

```diff
@@ -97,55 +98,63 @@ private function registerEventListeners(OutputInterface $output)
 
     /**
      * @param OutputInterface $output
-     * @param array $errorMessages
-     * @param array $warnings
+     * @param array           $errorMessages
+     *
+     * @return int
      */
-    private function printErrorMessages(OutputInterface $output, array $errorMessages, array $warnings)
+    private function returnErrorMessages(OutputInterface $output, array $errorMessages, array $warnings)
     {
-        $failed = $this->getPathsHelper()->getAsciiContent('failed');
+        $failed = $this->paths()->getAsciiContent('failed');
         if ($failed) {
             $output->writeln('<fg=red>' . $failed . '</fg=red>');
         }
 
-        $this->printWarningMessages($output, $warnings);
+        $this->returnWarningMessages($output, $warnings);
 
         foreach ($errorMessages as $errorMessage) {
             $output->writeln('<fg=red>' . $errorMessage . '</fg=red>');
         }
+
+        if (!$this->config->hideCircumventionTip()) {
+            $output->writeln(
+                '<fg=yellow>To skip commit checks, add -n or --no-verify flag to commit command</fg=yellow>'
+            );
+        }
+
+        return self::CODE_ERROR;
```

Comment:
I think it is a better design to let the caller manage the return code since this method's return value is invariant otherwise.